### PR TITLE
Add support for analog transmission range sensors

### PR DESCRIPTION
**Pull request by chuckwagoncomputing**

### CODE REVIEW

File: firmware/controllers/sensors/sensor_type.h [hunk 1]
```diff
@@ -113,6 +113,13 @@ enum class SensorType : unsigned char {
 
 	DetectedGear,
 
+	RangeInput1,
```
Comment (mck1117) [inline firmware/controllers/sensors/sensor_type.h:116]: does it make sense to put the un-decoded range inputs in the sensor model, or to put only the "decoded" PRNDL21whatever?

Comment (chuckwagoncomputing) [reply firmware/controllers/sensors/sensor_type.h:116]: It looked to me like the way to read an analog pin was to create a Sensor for it. Is there a better way?

Comment (chuckwagoncomputing) [reply firmware/controllers/sensors/sensor_type.h:116]: Also I found ResistanceFunc and wanted to use it.

**Comment by chuckwagoncomputing** [comment]
kinetis build failed with "cannot move location counter backwards (from 20008058 to 20008000)"
I may have deducted the wrong amount from `unusedOftenChangesDuringFirmwareUpdate`.

**Comment by chuckwagoncomputing** [comment]
Oh I see Kinetis has it's own RAM_UNUSED_SIZE, which is 1.
https://github.com/rusefi/rusefi/blob/master/firmware/config/boards/kinetis/efifeatures.h#L329

What do?

File: firmware/controllers/sensors/sensor_type.h (hunk 1 above)
Comment (chuckwagoncomputing) [reply firmware/controllers/sensors/sensor_type.h:116]: Andrey's suggestion was to reuse the Lua aux analog input Sensors, but that seemed like a bad idea to me.

**Comment by rusefillc** [comment]
> kinetis build failed with "cannot move location counter backwards"

I suggest exploring https://github.com/rusefi/rusefi/commit/12fb012321b31cfc234b032545fa6640dc3bb9d8 further if needed

File: firmware/controllers/sensors/sensor_type.h (hunk 1 above)
Comment (rusefillc) [reply firmware/controllers/sensors/sensor_type.h:116]: > Andrey's suggestion was to reuse the Lua aux analog input Sensors, but that seemed like a bad idea to me.

that was suggested only as a dev time shortcut

**Comment by chuckwagoncomputing** [comment]
Maybe init_range.cpp should get EFI_TCU wrapper

**Pull request merged**